### PR TITLE
Fetch latest trusted or verified block from the store as latest trusted state

### PR DESCRIPTION
**Pull request by romac**
See https://github.com/informalsystems/tendermint-rs/pull/375/files#r448951616 for context.

This PR also renames `Store::highest` to `Store::latest` for consistency with the spec.

* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGES.md

### CODE REVIEW

**Comment by romac** [comment]
I am not a big fan of this new `latest_trusted_or_verified` method, but it keeps the light client code short and clean. Ideally we'd like a way of combining `Status` values using boolean algebra to express AND/OR queries, but that seems a bit overkill at the moment.

**Comment by OStevan** [comment]
@romac thanks for the quick answer and follow up.

**Comment by romac** [comment]
I just updated the conformance test to use the `Trusted` status for the initial trusted block, which should prevent regressions (eg. we only ask for a "verified" block where we should ask for "verified or trusted").

File: light-client/src/store.rs [hunk 1]
```diff
@@ -20,15 +20,38 @@ pub mod sled;
...
     /// Overrides any other block with the same height and status.
     fn insert(&mut self, light_block: LightBlock, status: Status);
+
     /// Remove the light block with the given height and status, if any.
     fn remove(&mut self, height: Height, status: Status);
-    /// Get the highest light block with the given status.
-    fn highest(&self, status: Status) -> Option<LightBlock>;
+
+    /// Get the latest light block with the given status.
+    fn latest(&self, status: Status) -> Option<LightBlock>;
```
Comment (liamsi) [inline light-client/src/store.rs:35]: Just to remove any remaining ambiguity: previously this was called "highest" and it was pretty clear that this refers to the height. Latest is a bit unclear: latest added? Latest as in newest regarding block time, or, height? I think the doc comment could be explicit (the name is pr. fine though).

Comment (romac) [reply light-client/src/store.rs:35]: Good point! As in the spec, it means latest as in newest regarding height, will clarify in doc comment.

Comment (romac) [reply light-client/src/store.rs:35]: See https://github.com/informalsystems/tendermint-rs/pull/407/commits/dc63808b6f7b89ee37b7d6b837623757ea9ebda9

**Pull request merged**